test/start.test.js

```diff
@@ -548,14 +549,56 @@ test('should start the server with watch options that the child process restart
   await once(fastifyEmitter, 'ready')
   t.pass('should receive ready event')
 
-  await writeFile(tmpjs, 'hello fastify', { flag: 'a+' }) // chokidar watch can't caught change event in travis CI, but local test is all ok. you can remove annotation in local environment.
+  await writeFile(tmpjs, 'hello fastify', { flag: 'a+' }) // chokidar watch can't catch change event in CI, but local test is all ok. you can remove annotation in local environment.
   t.pass('change tmpjs')
 
   // this might happen more than once but does not matter in this context
   await once(fastifyEmitter, 'restart')
   t.pass('should receive restart event')
 })
 
+test('should reload the env on restart when watching', async (t) => {
```

Comment:
this test is similar to the existing one but I believe more robust, because it uses tap's native testdir to create a directory and change the cwd to that, so we're sure that the test is isolated from others. I know it introduces a different pattern from other tests, but I think it's better written in this way.